machine_learning/logistic_regression.py

```diff
@@ -42,11 +42,64 @@ def sigmoid_function(z):
 
     @param z:  input to the function
     @returns: returns value in the range 0 to 1
+
+    Examples:
+    >>> sigmoid_function(4)
+    0.9820137900379085
+    >>> sigmoid_function(np.array([-3,3]))
+    array([0.04742587, 0.95257413])
+    >>> sigmoid_function(np.array([-3,3,1]))
+    array([0.04742587, 0.95257413, 0.73105858])
     """
     return 1 / (1 + np.exp(-z))
 
 
 def cost_function(h, y):
+    """
```

Comment:
Can you add type hints.

Reply:
> Can you add type hints.

I'll add them in a different commit.

Reply:
You mean different PR?

Reply:
> You mean different PR?

Yes, my bad.

Reply:
> Can you add type hints.

I added the typehints as suggested by you